### PR TITLE
Refs [TASK][34069] Check Internet

### PR DESCRIPTION
**Pull request by trangnq258**
## Related Tickets
- [#Task 34069: Check Internet](https://edu-redmine.sun-asterisk.vn/issues/34069)
## WHAT
- Check Internet
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/yummy/utlis/ViewExtentions.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.yummy.utlis
+
+import android.view.View
+import com.example.yummy.R
+import com.google.android.material.snackbar.Snackbar
+
+fun View.make(message: String, length: Int = Snackbar.LENGTH_LONG) {
```
Comment (datdt-1935) [inline app/src/main/java/com/example/yummy/utlis/ViewExtentions.kt]: dùng tên gì có nghĩa hơn đi em

File: app/src/main/java/com/example/yummy/ui/search/SearchFragment.kt [hunk 2]
```diff
@@ -28,6 +29,11 @@ class SearchFragment : BaseFragment(), SearchContract.View {
         val context = context ?: return
         val repository = RepositoryUtils.getMealRepository(context)
         searchPresenter = SearchPresenter(this, repository)
+        isConnection = NetworkUtil.isConnection(context)
+        if (!isConnection) {
+            view?.make(getString(R.string.msg_check_internet))
+            return
+        }
```
Comment (datdt-1935) [inline app/src/main/java/com/example/yummy/ui/search/SearchFragment.kt]: nên để ở trong baseFragment vì mọi fragment đều dùng như nhau

File: app/src/main/res/values-vi/strings.xml [hunk 3]
```diff
@@ -53,4 +53,5 @@
     <string name="msg_delete_success">Xóa thành công</string>
     <string name="msg_insert_success_meal_favorite">Thêm %s thành công</string>
     <string name="msg_delete_success_meal_favorite">Xóa %s thành công</string>
+    <string name="msg_check_internet">Please check your internet connection!</string>
```
Comment (datdt-1935) [inline app/src/main/res/values-vi/strings.xml]: để tiếng việt đi em

File: app/src/main/java/com/example/yummy/ui/search/SearchFragment.kt (hunk 2 above)
Comment (trangnq258) [reply app/src/main/java/com/example/yummy/ui/search/SearchFragment.kt]: em không để trong baseFragment vì không phải fragment nào cũng cần check Internet (ví dụ như note ạ)

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**